esp/bindings/http/platform/httpservice.cpp

```diff
@@ -1120,6 +1125,208 @@ EspAuthState CEspHttpServer::preCheckAuth(EspAuthRequest& authReq)
     return authUnknown;
 }
 
+EspAuthState CEspHttpServer::verifyCookies(EspAuthRequest& authReq)
+{
+    CIArrayOf<CESPCookieVerification> cookies;
+    //Check whether the ESP request has specified which cookie to be verified or not.
+    IProperties* parms = m_request->queryParameters();
+    if (parms)
+    {
+        Owned<IPropertyIterator> iter = parms->getIterator();
+        ForEach(*iter)
+        {
+            const char* name = iter->getPropKey();
+            if (strieq(name, "__querystring"))
+                continue; //skip the property created by ESP.
+
+            //ESP client cannot access the session ID cookie (both cookie name and value).
+            //The client may use "ESPSessionID" to verify the session ID cookie.
+            if (strieq(name, "ESPSessionID"))
+                cookies.append(*(new CESPCookieVerification("ESPSessionID", "")));
```

Comment:
Why is this necessary?  You already know "name" is "ESPSessionID", except perhaps for the casing of that string. Can't you always call
    cookies.append(*(new CESPCookieVerification(name, "")));